test/test_foreach.py

```diff
@@ -87,25 +87,55 @@ def _test_unary_op(self, device, dtype, foreach_op, foreach_op_, torch_op):
 
     def _test_pointwise_op(self, device, dtype, foreach_op, foreach_op_, torch_op):
         for N in N_values:
-            tensors = self._get_test_data(device, dtype, N)
-            tensors1 = self._get_test_data(device, dtype, N)
-            tensors2 = self._get_test_data(device, dtype, N)
-            value = 2
+            values = [2 + i for i in range(N)]
+            for vals in [values[0], values]:
+                tensors = self._get_test_data(device, dtype, N)
+                tensors1 = self._get_test_data(device, dtype, N)
+                tensors2 = self._get_test_data(device, dtype, N)
 
-            # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
-            control_dtype = torch.float32 if (self.device_type == 'cuda' and
-                                              (dtype is torch.float16 or dtype is torch.bfloat16)) else dtype
-            expected = [torch_op(tensors[i].to(dtype=control_dtype),
-                                 tensors1[i].to(dtype=control_dtype),
-                                 tensors2[i].to(dtype=control_dtype), value=value).to(dtype=dtype) for i in range(N)]
+                # Mimics cuda kernel dtype flow.  With fp16/bf16 input, runs in fp32 and casts output back to fp16/bf16.
+                control_dtype = torch.float32 if (self.device_type == 'cuda' and 
+                                                  (dtype is torch.float16 or dtype is torch.bfloat16)) else dtype
 
-            res = foreach_op(tensors, tensors1, tensors2, value)
-            foreach_op_(tensors, tensors1, tensors2, value)
-            self.assertEqual(res, tensors)
-            if (dtype is torch.float16 or dtype is torch.bfloat16) and TEST_WITH_ROCM:
-                self.assertEqual(tensors, expected, atol=1.e-3, rtol=self.dtype_precisions[dtype][0])
-            else:
-                self.assertEqual(tensors, expected)
+                if vals == 2:
```

Comment:
`not isinstance(vals, list)` is more robust: someone could change `values` and then this could would never execute

Reply:
fiexd